pkg/scheduling/nodetemplate.go

```diff
@@ -27,22 +27,26 @@ import (
 // the fields in Provisioner. These structs are maintained separately in order
 // for fields like Requirements to be able to be stored more efficiently.
 type NodeTemplate struct {
+	ProvisionerName      string
```

Comment:
This is currently used by extracting the name from `labels` elsewhere in the code.

Reply:
Yeah I felt really dirty doing this. I think we might need another concept in scheduler.go that sits outside of NodeTemplate.